src/core/thread/mle_router.cpp

```diff
@@ -1309,7 +1310,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     case kDeviceStateChild:
         if ((mDeviceMode & ModeTlv::kModeFFD) && (GetActiveRouterCount() < mRouterUpgradeThreshold))
         {
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+            mRouterSelectionJitterTimer.Start(Timer::SecToMsec(otPlatRandomGet() % kRouterSelectionJitter));
```

Comment:
If the spec only indicates the delay should be applied when handling an advertisement message, then we should stick to that.  Can you indicate where in the spec the delay is limited to advertisement messages?